makelib/golang.mk

```diff
@@ -159,10 +159,10 @@ go.test.integration: $(GOJUNIT)
 	@cat $(GO_TEST_OUTPUT)/integration-tests.log | $(GOJUNIT) -set-exit-code > $(GO_TEST_OUTPUT)/integration-tests.xml || $(FAIL)
 	@$(OK) go test integration-tests
 
-go.lint: $(GOLINT)
-	@$(INFO) go lint
-	@$(GOLINT) -set_exit_status=true $(GO_PACKAGES) $(GO_INTEGRATION_TEST_PACKAGES) || $(FAIL)
-	@$(OK) go lint
+go.lint: $(GOLANGCILINT)
+	@$(INFO) golangci-lint
+	@$(GOLANGCILINT) run || $(FAIL)
```

Comment:
also which packages does it lint? the code being replaced with `golint` was passed `$(GO_PACKAGES)` and `$(GO_INTEGRATION_TEST_PACKAGES)`

Reply:
> does golangci-lint exist with an error code when something fails?

It does:

```bash
$ golangci-lint run --fast
# Many redacted linter warnings
 echo $?
1
```

Reply:
> which packages does it lint?

I believe it uses some magic to find all non-vendor Go code under `$PWD`. I briefly tried passing it `$(GO_PACKAGES)` with no success, but didn't look into it in great detail. What are the advantages of passing in this precomputed list of packages vs letting the linter tool work it out?

Reply:
I think it would avoid scanning vendor directory, etc.

Reply:
So `golangci-lint` can take a list of paths to lint, but it expects them to be relative to `$PWD` (e.g. `pkg/...`) rather than fully qualfied (e.g. `github.com/crossplaneio/crossplane/pkg/...`). $(GO_PACKAGES)` is a list of the latter.

By default `golangci-lint run` will lint all files under its working directly. It provides `--skip-dirs` and `--skip-files` flags we could use to avoid linting certain paths. The default set of dirs includes `vendor/`, `testdata/`, etc - https://github.com/golangci/golangci-lint/blob/3345c71/pkg/packages/skip.go#L18

I'd prefer to go with the path of least resistance and simply run `golangci-lint run` without any paths until such time as we have a compelling reason to explicitly feed in paths.

Reply:
sounds good.